pandas/tests/reshape/test_concat.py

```diff
@@ -2730,3 +2730,13 @@ def test_concat_datetimeindex_freq():
     expected = pd.DataFrame(data[50:] + data[:50], index=dr[50:].append(dr[:50]))
     expected.index._data.freq = None
     tm.assert_frame_equal(result, expected)
+
+
+def test_concat_sparse():
+    # GH 23557
+    a = pd.Series(pd.SparseArray([0, 1, 2]))
+    expected = pd.DataFrame(data=[[0, 0], [1, 1], [2, 2]]).astype(
+        pd.SparseDtype("int", 0)
```

Comment:
```suggestion
        pd.SparseDtype(np.intp, 0)
```

Reply:
Thanks for the suggestion, I've amended the PR with the change

Reply:
I had to change this to `np.int64` else the tests were failing on linux 32-bit